### PR TITLE
refactor: use State for building database streams

### PR DESCRIPTION
**Pull request by sborrazas**

### CODE REVIEW

File: lib/ae_mdw/collection.ex [hunk 1]
```diff
@@ -55,16 +56,24 @@ defmodule AeMdw.Collection do
   Builds a stream from records from a table starting from the initial_key given.
   """
   @spec stream(table(), direction(), key_boundary(), cursor()) :: Enumerable.t()
-  def stream(tab, direction, key_boundary, cursor) do
+  def stream(tab, direction, key_boundary, cursor),
+    do: stream(State.new(), tab, direction, key_boundary, cursor)
```
Comment (jyeshe) [inline lib/ae_mdw/collection.ex:60]: Could you document the logic behind streaming from an empty state please?

File: lib/ae_mdw/collection.ex [hunk 2]
```diff
@@ -73,9 +82,15 @@ defmodule AeMdw.Collection do
   Streams forward a table seeking the iterator to a boundary start key.
   """
   @spec stream(table(), key()) :: Enumerable.t()
-  def stream(table, boundary_start_key) do
-    case fetch_first_key(table, :forward, boundary_start_key, nil) do
-      {:ok, first_key} -> unfold_stream(table, :forward, first_key, nil)
+  def stream(table, boundary_start_key), do: stream(State.new(), table, boundary_start_key)
```
Comment (jyeshe) [inline lib/ae_mdw/collection.ex:85]: same for this case, thank you

**Review by jyeshe — COMMENTED** [review]
Looks good. One documentation request only.

File: lib/ae_mdw/collection.ex (hunk 2 above)
Comment (sborrazas) [reply lib/ae_mdw/collection.ex:91]: Done for both

**Review by jyeshe — APPROVED** [review]
:rocket:

**Review by thepiwo — APPROVED** [review]

**Pull request merged**